pkg/kubelego/kubelego.go

```diff
@@ -297,20 +293,11 @@ func (kl *KubeLego) paramsLego() error {
 		kl.legoSupportedIngressProvider = strings.Split(supportedProviders, ",")
 	}
 
-	legoDefaultIngressProvider := os.Getenv("LEGO_DEFAULT_INGRESS_PROVIDER")
-	if len(legoDefaultIngressProvider) == 0 {
-		kl.legoDefaultIngressProvider = "nginx"
-	} else {
-		var err error = nil
-		kl.legoDefaultIngressProvider, err = ingress.IsSupportedIngressProvider(kl.legoSupportedIngressProvider, legoDefaultIngressProvider)
-		if err != nil {
-			return fmt.Errorf("Unsupported default ingress provider: '%s'. You can set the ingress provider with 'LEGO_DEFAULT_INGRESS_PROVIDER'", legoDefaultIngressProvider)
-		}
-	}
-
-	kl.legoSupportedIngressClass = strings.Split(os.Getenv("LEGO_SUPPORTED_INGRESS_CLASS"), ",")
-	if len(kl.legoSupportedIngressClass) == 0 {
```

Comment:
The issue here is that `len(strings.Split("", ",")) == 1`, and not 0, meaning the default would never be set.